common/workunit/CMakeLists.txt

```diff
@@ -78,5 +79,6 @@ if(NOT PLUGIN)
         nbcd 
         eclrtl 
         deftype
+        archivefile
```

Comment:
libarchive is not available on all platforms - it's not ok to make workunit dll dependent on it.

At the very least you need to make this code dependent on the USE_LIBARCHIVE cmake flag

Reply:
(an alternative approach would be to pipe("unzip"), or get the user to unzip before calling your code).

Reply:
I guess I have to support both libarchive and mannually unzip using USE_LIBARCHIVE.

Reply:
It may not be worth going down the libarchive route if have to implement using unzip anyway.
Or, could be a future JIRA/improvement later.